### PR TITLE
activation functions added for bounded outputs

### LINKED ISSUE #13: Activation functions for bounding the output of the model
### Is your feature request related to a problem? Please describe.

The output of some specific variables, total precipitation for example, is not bounded. Therefore, the model sometimes outputs negative values, which is not physical.

### Describe the solution you'd like

Adding a bounding strategy to the model via the usage of activation functions.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Organisation

ECMWF

### PR DESCRIPTION
**Pull request by gabrieloks**
This PR is related to the open issue https://github.com/ecmwf/anemoi-models/issues/13 and the aifs-mono PR https://github.com/ecmwf-lab/aifs-mono/pull/224.

The basic idea behind this PR is to implement bounding strategies for specific variables, chosen by the user, via the usage of activation functions. One evident example the total precipitation accumulation variable, which is not currently bounded, therefore prone to negative value predictions.

Closes #13

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](https://bol-claassistant.ecmwf.int:443/ecmwf/anemoi-models?pullRequest=14) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bol-claassistant.ecmwf.int:443/ecmwf/anemoi-models?pullRequest=14) before we can accept your contribution.<br/><hr/>**Gabriel Moldovan** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bol-claassistant.ecmwf.int:443/check/ecmwf/anemoi-models?pullRequest=14) it.</sub>

File: src/anemoi/models/preprocessing/normalizer.py [hunk 1]
```diff
@@ -40,6 +40,18 @@ def __init__(
         data_indices : dict
             Data indices for input and output variables
         """
+        # The only rapid way I found to avoid the _validate_normalization_inputs assert...
```
Comment (mchantry) [inline src/anemoi/models/preprocessing/normalizer.py:43]: @JesperDramsch do you have any ideas for improving this?

File: src/anemoi/models/models/encoder_processor_decoder.py [hunk 2]
```diff
@@ -103,6 +105,13 @@ def __init__(
             dst_grid_size=self._data_grid_size,
         )
 
+        self.boundings = nn.ModuleList(
```
Comment (mchantry) [inline src/anemoi/models/models/encoder_processor_decoder.py:108]: Can you add a comment to explain the these lines please?
e.g.
# Create the bounding objects (e.g. to ensure TP is positive definite)

Comment (gabrieloks) [reply src/anemoi/models/models/encoder_processor_decoder.py:109]: Done

File: src/anemoi/models/preprocessing/__init__.py [hunk 3]
```diff
@@ -47,8 +63,10 @@ def __init__(
         self.data_indices = data_indices
 
     def _process_config(self, config):
```
Comment (mchantry) [inline src/anemoi/models/preprocessing/__init__.py:65]: Can you put a comment to explain why these keys are handled specially?

**Comment by mchantry** [comment]
Great work on this, thanks.

File: src/anemoi/models/preprocessing/__init__.py (hunk 3 above)
Comment (JesperDramsch) [reply src/anemoi/models/preprocessing/__init__.py:65]: ok

**Review by mchantry — APPROVED** [review]
Great work, approved. Please merge.

**Review by JesperDramsch — APPROVED** [review]

**Pull request merged**